Online/RPCs.cs

```diff
@@ -121,49 +121,83 @@ public static void GoToDeathScreen()
+                    && playerAvatar.FindEntity(true) is OnlinePhysicalObject opo
+                    && opo.apo is AbstractCreature ac
+                    && ac.Room.shelter)
+                {
+                    denPos = ac.Room.name;
+                }
+            }
+
+            if (denPos != null)
+            {
+                game.GetStorySession.saveState.denPosition = denPos;
+                (OnlineManager.lobby.gameMode as StoryGameMode).defaultDenPos = denPos;
+            }
+
             foreach (OnlinePlayer player in OnlineManager.players)
             {
-                if (!player.OutgoingEvents.Any(e => e is RPCEvent rpc && rpc.IsIdentical(RPCs.GoToWinScreen, malnurished)))
+                if (!player.OutgoingEvents.Any(e => e is RPCEvent rpc && rpc.IsIdentical(RPCs.GoToWinScreen, malnourished, denPos)))
                 {
-                    player.InvokeRPC(RPCs.GoToWinScreen, malnurished);
+                    if (player.isMe)
+                    {
+                        GoToWinScreen(malnourished, denPos);
+                    }
+                    else
+                    {
+                        player.InvokeRPC(RPCs.GoToWinScreen, malnourished, denPos);
+                    }
                 }
             }
         }
 
         //Assumed to be called for storymode only
         [RPCMethod]
-        public static void GoToWinScreen(bool malnourished)
+        public static void GoToWinScreen(bool malnourished, string denPos)
         {
+            RainMeadow.Debug($"({malnourished})");
             var game = (RWCustom.Custom.rainWorld.processManager.currentMainLoop as RainWorldGame);
             if (game == null || game.manager.upcomingProcess != null)
             {
                 return;
             }
 
-            if (!malnourished && !game.rainWorld.saveBackedUp)
+            if (OnlineManager.lobby.isOwner)
             {
-                game.rainWorld.saveBackedUp = true;
-                game.rainWorld.progression.BackUpSave("_Backup");
+                if (!malnourished && !game.rainWorld.saveBackedUp)
+                {
+                    game.rainWorld.saveBackedUp = true;
+                    game.rainWorld.progression.BackUpSave("_Backup");
+                }
             }
-
-            //This needs to be called after shelterDoor::Close and game state synced for this to be accurate.
-            var denPos = (OnlineManager.lobby.gameMode as StoryGameMode).defaultDenPos;
-            if (denPos == null) {
-                AbstractCreature firstAlivePlayer = game.FirstAlivePlayer;
-                denPos = game.world.GetAbstractRoom(firstAlivePlayer.pos).name;
+            else
+            {
+                var storyClientSettings = OnlineManager.lobby.gameMode.clientSettings as StoryClientSettings;
+                if (storyClientSettings.isDead)
+                {
+                    storyClientSettings.myLastDenPos = denPos;
```

Comment:
Might be mistaken, but are we moving dead players to the hosts hibernation den?

Reply:
yup, host passes their den position thru GoToWinScreen